### ISSUE TITLE
Generated Images Not Appearing

### ISSUE THREAD

**Issue by IsaccBarker** [astriaai/headshots-starter#123]
Hello, I uploaded some photos of myself and paid for a credit because I was curious what sort of images it would give me. I wasn't expecting no images! The training data column is populated, but the results one is not. Am I missing something obvious?

**Comment by bear75** [comment]
Same for me, however I see the result images in astria…

**Comment by AbhishekTyagi0** [comment]
Yes, there is an issue with the webhooks between astria and the headshots starter.

I thought it was only happening to me only.

@burgalon Hi please can you look into it, I have given so much time building my app, and then this issue arose.

**Comment by burgalon** [comment]
@IsaccBarker @bear75 @AbhishekTyagi0 
Please go into your Vercel console, project, settings, environment variable and make sure VERCEL_URL is set to a public domain that does not require authentication
You can see the callback url that gets sent to Astria if you open https://www.astria.ai/tunes.json
Make sure to test things using astriaTestMode / ASTRIA_TEST_MODE turned on to avoid charges and also fast processing time.

![image](https://github.com/user-attachments/assets/015984ec-b397-4be8-a72e-af4dae33ad90)

**Comment by burgalon** [comment]
If this doesn't work for some reason, you can just change the code so that [callback](https://github.com/astriaai/headshots-starter/blob/5fbe10b3a65c9df75cd90266f2e13246750245be/app/astria/train-model/route.ts#L147C49-L147C59) does not using VERCEL_URL and is instead hard-coded to the desired domain.

**Comment by IsaccBarker** [comment]
> @IsaccBarker @bear75 @AbhishekTyagi0 Please go into:
> 
>     1. go to https://vercel.com/
> 
>     2. select project
> 
>     3. Settings tab
> 
>     4. Select Environment variables on the left
> 
>     5. make sure VERCEL_URL is set to a public domain that does not require authentication, and is not getting redirects - e.g: http => https, or non www => wwww - the callback client does not follow redirect
> 
> 
> You can see the callback url that gets sent to Astria if you open https://www.astria.ai/tunes.json
> 
> Make sure to test things using astriaTestMode / ASTRIA_TEST_MODE turned on to avoid charges and also fast processing time.
> 
> ![image](https://private-user-images.githubusercontent.com/47217/375325391-015984ec-b397-4be8-a72e-af4dae33ad90.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1NzgwOTksIm5iZiI6MTcyODU3Nzc5OSwicGF0aCI6Ii80NzIxNy8zNzUzMjUzOTEtMDE1OTg0ZWMtYjM5Ny00YmU4LWE3MmUtYWY0ZGFlMzNhZDkwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDE2Mjk1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyMzlhOWEwYTk0NTA4YTE1NmYxYTFhYmNhOWQ5N2U3NzhmMTBjMjk1MWFhZTk1NTQ3ZGMyOWZmOWY0ZWJkNmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VBRwZY4uhHC5-x3pGP-pObA9abOnTBmkQTJtwRE1dzc)

I didn't realize this was a template repository, so I uploaded my photos to the example hosted by this repository. Since I'm not hosting this, I don't have the option to change the Vercel settings.

**Comment by AbhishekTyagi0** [comment]
Thank all I was able to solve this issue with the help of dear Alon sir aka @burgalon

**Comment by burgalon** [comment]
Did you check the vercel server logs ?

**Comment by kevincolten** [comment]
You can copy/paste https://github.com/astriaai/headshots-starter/blob/main/supabase/migrations/20231010160942_remote_schema.sql into the Supabase Query Editor to run the migration

**Comment by kevincolten** [comment]
When using the VERCEL_URL environment variable, I had to disable "Automatically expose System Environment Variables", as it seemed use prefer its own internal VERCEL_URL over mine.

![Image](https://github.com/user-attachments/assets/b9e38adc-9e85-4e59-b275-d9ba3ef7d2fb)

**Comment by kevincolten** [comment]
You also have to make sure JSON callback is enabled in your Astria API settings

![Image](https://github.com/user-attachments/assets/4bef4afd-6374-4c46-b44e-ce357169cf5a)
